src/utils/specs.py

```diff
@@ -34,67 +34,106 @@
+    L2Banks: str
+    LDSBanks: str
+    numSQC: str
+    hbmBW: str
+    compute_partition: str
+    memory_partition: str
 
     def __str__(self):
         return dedent(
             f"""\
         Host info:
             hostname:       {self.hostname}
-            cpu info:       {self.cpu}
+            CPU:            {self.CPU}
+            sbios:          {self.sbios}
             ram:            {self.ram}
             distro:         {self.distro}
-            kernel version: {self.kernel}
-            rocm version:   {self.rocmversion}
+            kernel_version: {self.kernel_version}
+            rocm_version:   {self.rocm_version}
         Device info:
             GPU:                {self.GPU}
-            L1:                 {self.L1}
-            L2:                 {self.L2}
-            Max SCLK:           {self.SCLK}MHz
-            Current SCLK:       {self.cur_SCLK}MHz
-            Current MCLK:       {self.cur_MCLK}MHz
+            arch:               {self.arch}
+            vbios:              {self.vbios}
+            L1:                 {self.L1} KB
+            L2:                 {self.L2} KB
+            max_sclk:           {self.max_sclk} MHz
+            cur_sclk:           {self.cur_sclk} MHz
+            cur_mclk:           {self.cur_mclk} MHz
             CU:                 {self.CU}
             SIMD:               {self.SIMD}
             SE:                 {self.SE}
-            Wave Size:          {self.wave_size}
-            Workgroup Max Size: {self.workgroup_size}
-            Max Wave Occupancy Per CU: {self.wave_occu}
+            wave_size:          {self.wave_size}
+            workgroup_max_size: {self.workgroup_max_size}
+            max_waves_per_cu:   {self.max_waves_per_cu}
+            L2Banks:            {self.L2Banks}
+            LDSBanks:           {self.LDSBanks}
+            numSQC:             {self.numSQC}
+            hbmBW:              {self.hbmBW} MB/s
+            compute_partition:  {self.compute_partition}
+            memory_partition:   {self.memory_partition}
         """
         )
 
 
 def gpuinfo():
-    rocminfo = run(["rocminfo"]).split("\n")
+    # Local var only for rocminfo searching
+    gpu_list = {"gfx906", "gfx908", "gfx90a", "gfx940", "gfx941", "gfx942"}
+
+    # Fixme: find better way to differentiate cards, GPU vs APU, etc.
+    rocminfo_full = run(["rocminfo"])
+    rocminfo = rocminfo_full.split("\n")
 
     for idx1, linetext in enumerate(rocminfo):
-        gpu_id = search(r"^\s*Name\s*:\s+ ([a-zA-Z0-9]+)\s*$", linetext)
-        if gpu_id in gpu_list:
+        gpu_arch = search(r"^\s*Name\s*:\s+ ([a-zA-Z0-9]+)\s*$", linetext)
+        if gpu_arch in gpu_list:
             break
-
-    if not gpu_id in gpu_list:
-        return None, None, None, None, None, None, None, None, None, None
+        if str(gpu_arch) in gpu_list:
+            gpu_arch = str(gpu_arch)
+            break
+    if not gpu_arch in gpu_list:
+        return (
+            None,
```

Comment:
Have to admit, not crazy about returning a very specific number of "None"s here.  Strikes me that the configuration we want to capture from `rocminfo` might change over time (or for newer SoCs) and the ordering could be problematic.  Should we consider creating and returning a dict to be more organized here?


```
gpuFoo['L1'] = xx
gpuFoo['L2'] = yy
...
gpuFoo['wave_size'] = zz
```

Reply:
I agree. It's now converted to a dictionary.
https://github.com/AMDResearch/omniperf/blob/cbbd4fe75eeda296e5e4b083a5019a8a399e12e5/src/utils/specs.py#L109-L126